MyMusic/src/main/java/com/ciandt/summit/bootcamp2022/domains/songs/adapters/services/SongServiceImp.java

```diff
@@ -1,6 +1,45 @@
 package com.ciandt.summit.bootcamp2022.domains.songs.adapters.services;
 
+import com.ciandt.summit.bootcamp2022.domains.exceptions.songs.InvalidSongNameOrArtistNameException;
+import com.ciandt.summit.bootcamp2022.domains.exceptions.songs.SongsNotFoundException;
+import com.ciandt.summit.bootcamp2022.domains.songs.Song;
+import com.ciandt.summit.bootcamp2022.domains.songs.dtos.SongDTO;
 import com.ciandt.summit.bootcamp2022.domains.songs.ports.interfaces.SongServicePort;
+import com.ciandt.summit.bootcamp2022.domains.songs.ports.repositories.SongRepositoryPort;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class SongServiceImp implements SongServicePort {
+    private final SongRepositoryPort songRepositoryPort;
+
+    public SongServiceImp(SongRepositoryPort songRepositoryPort) {
+        this.songRepositoryPort = songRepositoryPort;
+    }
+
+    @Override
+    public List<SongDTO> findByNameOrArtistName(String name, int pageNumber)
+            throws InvalidSongNameOrArtistNameException, SongsNotFoundException {
+
+        if(name == null || name.isBlank() || name.length() < 2){
+            throw new InvalidSongNameOrArtistNameException("Parameter must be at least 2 characters long.");
```

Comment:
pode ser interessante chamar de "filter" ao invés de "parameter", assim se essa mensagem de erro for retornada, fica mais coeso